src/base/emu-i386/kvm.c

```diff
@@ -158,6 +158,76 @@ static struct kvm_userspace_memory_region maps[MAXSLOT];
 
 static int init_kvm_vcpu(void);
 
+#if !defined(DISABLE_SYSTEM_WA) || !defined(KVM_CAP_IMMEDIATE_EXIT)
+
+/* compat functions for older complex method of immediate exit
+   using a signal that is blocked outside KVM_RUN but not blocked inside it */
+
+#include <pthread.h>
+
+#ifndef KVM_CAP_IMMEDIATE_EXIT
+#define KVM_CAP_IMMEDIATE_EXIT 136
+#define immediate_exit padding1[0]
+#endif
+
+#define KVM_IMMEDIATE_EXIT_SIG (SIGRTMIN + 1)
+static int kvm_cap_immediate_exit = -1;
+
+static void kvm_compat_set_immediate_exit(int set)
+{
+  if (kvm_cap_immediate_exit == -1) { // setup
+    struct sigaction sa;
+    sigset_t set;
```

Comment:
Oops, you got a parameter shadowing here!
I believe this code is not tested, or what am
I missing?

Reply:
it was tested, but shadowing is not good in any case, that's for sure!